Do not ignore error from gssapi InitSecContext

EugeneChung opened the pull request:
- but we must consider ErrContinueNeeded as non-error like EWOULDBLOCK/EAGAIN of read/write system calls

Commit:
Do not ignore error from gssapi InitSecContext

- but we must consider ErrContinueNeeded as normal like EWOULDBLOCK/EAGAIN of read/write system calls

beltran reviewed (commented):
Test is failing, you will have to comment the failing lines like in the previous cls

Commit:
Comment out calling client.Step()

After making it return the error from initClientContext() of gssapi.go properly, calling client.Step() seems always fail here.

beltran commented on gssapi.go:
```diff
@@ -55,7 +55,9 @@ func (m *GSSAPIMechanism) step(challenge []byte) ([]byte, error) {
 		err := initClientContext(m.context, m.service+"/"+m.host, challenge)
 		if err != nil {
-			log.Fatal(err)
+			if err != gssapi.ErrContinueNeeded {
```
I think this would still have the same problem as before. We should a single if with something like:

```
                if err != nil && err != gssapi.ErrContinueNeeded {
			log.Fatal(err)
			return nil, err
		}
``

Commit:
check error from initClientContext() properly
